mono/metadata/threads.c

```diff
@@ -6546,7 +6547,7 @@ mono_threads_summarize_execute_internal (MonoContext *ctx, gchar **out, MonoStac
 {
 	static SummarizerGlobalState state;
 
-	int current_idx;
+	int current_idx = 0;
```

Comment:
The logic here is not clear. One use of ‘current_idx’ is guarded by return value of function that sets it, and one is not.

I can remove this commit for now.

@kyte

Reply:
Current thread should probably be special case, so current_idx always set.
If it isn't found, abort the dump.
This fixes the case of >128 threads and current thread isn't in the first 128.
Which seems easy enough to produce.